src/commonMain/kotlin/io/modelcontextprotocol/kotlin/sdk/server/Server.kt

```diff
@@ -156,9 +130,104 @@ public open class Server(
         }
     }
 
+    @Deprecated(
+        "Will be removed with Protocol inheritance. Use connectSession instead.",
+        ReplaceWith("connectSession"),
+        DeprecationLevel.WARNING
+    )
+    public override fun onClose() {
+        logger.debug { "Closing MCP server" }
+        _onClose()
+    }
+
+    // TODO: Rename closeSessions to close after the full onClose deprecation
+    public suspend fun closeSessions() {
+        logger.debug { "Closing MCP server" }
+        sessions.value.forEach { it.close() }
+        _onClose()
+    }
+
+    /**
+     * Starts a new server session with the given transport and initializes
+     * internal request handlers based on the server's capabilities.
+     *
+     * @param transport The transport layer to connect the session with.
+     * @return The initialized and connected server session.
+     */
+    // TODO: Rename connectSession to connect after the full connect deprecation
+    public suspend fun connectSession(transport: Transport): ServerSession {
```

Comment:
It should be renamed to connect after several deprecation cycles, but now I can not name it like this due to `Protocal` inheritance